### DISCUSSION TITLE
Word-Level-Timestamps missing latter half of transcription

### DISCUSSION THREAD

**Discussion by jafrizzell** [openai/whisper#1104] [Q&A] [answered]
Hi,

I was using Whisper with `word_level_timestamps=True`, and noticed that the model would frequently show more text in the `text` key compared to the `segments[0]['words']` key.

Example:
If the `transcript` of an audio file was a recording of "The quick brown fox jumped over the lazy dog", the `transcript['text']` would accurately reflect `transcript['text'] = "The quick brown fox jumped over the lazy dog"`.

However, the `transcript['segments'][0]['words']` would look like this (removed confidence key for conciseness):
`[['word': 'The', 'start': 0.0, 'end': 1.0], ['word': 'quick', 'start': 1.0, 'end': 2.0], ['word': 'brown', 'start': 2.0, 'end': 3.0], ['word': 'fox', 'start': 3.0, 'end': 4.0]]` 
which is of course missing the words "jumped over the lazy dog"

Is this an issue of hallucination, or something else? Has anyone else seen this issue when using word-level-timestamps?

**Comment by jongwook** [comment, marked as answer]
Hi, are you by any chance using an old commit in the PR? Asking because I renamed the option to `word_timestamps` in the merged version, and the `["words"]` list also includes per-word probabilities. The merged version also fixed some issues similar to what you're seeing. Please try:

```
pip uninstall whisper openai-whisper
pip install openai-whisper==20230314
```

**Reply by jafrizzell** [reply]
Upgrading to the latest version helped, but the issue still occurs sometimes:

`transcript['segments'][0]['words'] = [{'word': ' Wait,', 'start': 5.7, 'end': 6.02, 'probability': 0.6553437113761902}, {'word': " that's", 'start': 0.8, 'end': 6.34, 'probability': 0.9782531261444092}, {'word': ' easy.', 'start': 6.34, 'end': 6.48, 'probability': 0.9946032166481018}]`

`transcript['text'] = Wait, that's easy. Alright, we don't have to skip this, bro.`

**Reply by ryanheise** [reply]
Do you have an example mp3 (preferably short)?

**Reply by jafrizzell** [reply]
Unfortunately my application is using Twitch livestreams so I don't have any data downloaded. You can see the code in my repo [moonlights-transcribe](https://github.com/jafrizzell/moonlights-transcribe) (I haven't pushed any changes involving word-level-timestamps yet).

If you want to try and replicate it you can clone the repo and run it locally. You don't need a Twitch API key if you only run on streams that are currently live, and remove the lines that reference API calls.

**Reply by ryanheise** [reply]
I guess that would make it difficult to investigate.

I have found other examples where both text and words are missing from the end of a segment which I'm currently investigating, but that's not the same as what you describe.

If there is a mismatch, I suspect it would be related to some recent work here: #1087 .
